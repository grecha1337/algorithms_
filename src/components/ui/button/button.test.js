import { Button } from "./button";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("Тестирование кнопки", () => {
  it("кнопка с текстом", () => {
    const button = renderer.create(<Button text="text" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("кнопка без текстом", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("заблокированной кнопки", () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("кнопки с индикацией загрузки.", () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("корректность вызова колбека при клике на кнопку", () => {
    const mockFn = jest.fn();
    render(<Button data-testid="button" onClick={mockFn} />);
    const button = screen.getByTestId("button");
    fireEvent.click(button);
    expect(mockFn).toHaveBeenCalled();
  });
});
