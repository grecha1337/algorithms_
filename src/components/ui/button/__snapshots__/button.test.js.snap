// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование кнопки заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование кнопки кнопка без текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование кнопки кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Тестирование кнопки кнопки с индикацией загрузки. 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
